### DISCUSSION TITLE
How to execute this trim query as an sqlalchemy statement ?

### DISCUSSION THREAD

**Discussion by shyam-sreenivasan** [sqlalchemy/sqlalchemy#8428] [Usage Questions]
### Describe the use case

I would like to translate this MySQL query to sqlalchemy construct.
select TRIM(LEADING '0' FROM myField);

How do I do this with func.trim()

### Databases / Backends / Drivers targeted

MySQL

### Example Use

select TRIM(LEADING '0' FROM myField);

**Comment by zzzeek** [comment]
we don't have a built in for mysql.TRIM() right now though one could be added.    you phrased your issue as a question so I assume this was a usage question, not an official feature request where you were also sending a PR.

to render arbitrary SQL expressions, we generally use literal_column(), which has a lot of ways to build things like this

straight:

```py
>>> from sqlalchemy import literal_column, select
>>> print(select(literal_column("TRIM(LEADING '0' FROM myField)")))
SELECT TRIM(LEADING '0' FROM myField)
```

using func on the outside:

```py
>>> from sqlalchemy import func
>>> print(select(func.TRIM(literal_column("LEADING '0' FROM myField)"))))
SELECT TRIM(LEADING '0' FROM myField)) AS "TRIM_1"

```

if we look more closely at MySQL's syntax, "FROM" is really an operator here, so we can combine literal_column with op() so that you can get arbitrary expressions as the field, probably use this for now

```py
>>> from sqlalchemy import column
>>> col = column("mycol")
>>> print(select(func.TRIM(literal_column("LEADING '0'").op("FROM")(col))))
SELECT TRIM(LEADING '0' FROM mycol) AS "TRIM_1"

```
